src/status_im/subs/wallet/screens/input_amount.cljs

```diff
@@ -0,0 +1,360 @@
+
+(rf/reg-sub :send-input-amount-screen/upper-limit-equals-input-value?
+ :<- [:send-input-amount-screen/state]
+ :<- [:send-input-amount-screen/upper-limit]
+ (fn [[{:keys [input-value]}
+       upper-limit]]
+   (money/equal-to
+    (money/bignumber input-value)
+    (money/bignumber upper-limit))))
+
+
+(defn- fiat->crypto
+  [value conversion-rate]
+  (-> value
+      (money/fiat->crypto conversion-rate)
+      (utils/cut-crypto-decimals-to-fit-usd-cents conversion-rate)))
+
+(defn- crypto->fiat
+  [value conversion-rate]
+  (-> value
+      (money/crypto->fiat conversion-rate)
+      (utils/cut-fiat-balance-to-two-decimals)))
+
+(rf/reg-sub :send-input-amount-screen/input-value-converted
+ :<- [:send-input-amount-screen/state]
+ :<- [:send-input-amount-screen/conversion-rate]
+ (fn [[{:keys [crypto-currency? input-value]}
+       conversion-rate]]
+   (if crypto-currency?
+     (crypto->fiat input-value conversion-rate)
+     (fiat->crypto input-value conversion-rate))))
+
+(rf/reg-sub :send-input-amount-screen/converted-value-prettified
+ :<- [:send-input-amount-screen/state]
+ :<- [:send-input-amount-screen/input-value-converted]
+ :<- [:profile/currency-symbol]
+ :<- [:wallet/wallet-send-token]
+ (fn [[{:keys [crypto-currency?]}
+       token-input-converted-value
+       currency-symbol
+       {token-symbol :symbol}]]
+   (if crypto-currency?
+     (utils/prepend-curency-symbol-to-fiat-balance token-input-converted-value currency-symbol)
+     (utils/add-token-symbol-to-crypto-balance token-input-converted-value
+                                               (or (clj->js token-symbol) "")))))
+
+(rf/reg-sub :send-input-amount-screen/amount-in-crypto
+ :<- [:send-input-amount-screen/state]
+ :<- [:send-input-amount-screen/conversion-rate]
+ :<- [:send-input-amount-screen/token-decimals]
+ (fn [[{:keys [crypto-currency? input-value]}
+       conversion-rate
+       token-decimals]]
+   (if crypto-currency?
+     input-value
+     (number/remove-trailing-zeroes
+      (.toFixed (/ input-value conversion-rate) token-decimals)))))
+
+(rf/reg-sub :send-input-amount-screen/recipient-gets-amount
+ :<- [:send-input-amount-screen/conversion-rate]
+ :<- [:wallet/total-amount-in-to-chains]
+ :<- [:wallet/wallet-send-token]
+ (fn [[conversion-rate
+       total-amount-receiver
+       {token-symbol :symbol}]]
+   (utils/prettify-crypto-balance
+    token-symbol
+    total-amount-receiver
+    conversion-rate)))
+
+(rf/reg-sub :send-input-amount-screen/routes
+ :<- [:wallet/wallet-send-suggested-routes]
+ (fn [suggested-routes]
+   (when suggested-routes
+     (or (:best suggested-routes) []))))
+
+(defn- every-network-value-is-zero?
+  [sender-network-values]
+  (every? (fn [{:keys [total-amount]}]
+            (and
+             total-amount
+             (money/bignumber? total-amount)
+             (money/equal-to total-amount
+                             (money/bignumber "0"))))
```

Comment:
Bignums are not cheap and here it's re-created on every iteration. A common approach I've seen in other project is to declare the zero value as a static var inside `money` and declare it as a `^:const`.